src/dune_engine/install.mli

```diff
@@ -112,6 +112,12 @@ module Entry_with_site : sig
     }
 end
 
+module Metadata : sig
+  type 'src t =
+    | DefaultEntry of 'src Entry.t
+    | UserDefinedEntry of 'src Entry.t
```

Comment:
I wonder if this could be just a `defined_by : Default | User of loc` field in `Entry.t`?

Reply:
The Entry.t type is used to represent parsed .install file. So this `defined_by` field would need to be optional or some junk value in this case.

Still the current representation is a little too generic and inconvenient, I'm thinking of replacing it with:
```
module Entry : sig
  module With_source : sig
    type source = Dune | User
    type nonrec t =
       { source : source
       ; entry : t
      }
  end
end
```